File: ios/Classes/MyPackagePlugin.m
#import "MyPackagePlugin.h"
#import "GoogleMaps/GoogleMaps.h"
@implementation MyPackagePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"my_package"
            binaryMessenger:[registrar messenger]];
  MyPackagePlugin* instance = [[MyPackagePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  }
  else if([@"setGoogleMapKey" isEqualToString:call.method]){

  }
  else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: ios/Classes/MyPackagePlugin.swift
import Flutter
import UIKit
import GoogleMaps

public class MyPackagePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "my_package", binaryMessenger: registrar.messenger())
    let instance = MyPackagePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)

    case "setGoogleMapKey":
     if let jsonResult = call.arguments as? Dictionary<String, AnyObject> {
                        let googleMapsApiKey =  jsonResult["mapKey"] as? String ?? ""
                        GMSServices.provideAPIKey(googleMapsApiKey)
                    }
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
